astropy/io/fits/tests/test_fitscheck.py

```diff
@@ -92,3 +100,29 @@ def test_overwrite_invalid(self, caplog):
 
         # check that the file was fixed
         assert fitscheck.main([testfile]) == 0
+
+    def test_missing_invalid(self, caplog):
+        """
+        from https://github.com/astropy/astropy/issues/16551
+        written by Zach Claytor
+        """
+        # Test with a file where primary HDU has no checksum/datasum and image
+        # extension contains valid checksum/datasum
+        testfile = self.temp("test.fits")
+        hdul = fits.HDUList([fits.PrimaryHDU()])
+        ext = fits.ImageHDU(data=np.arange(10))
+        ext.header["THINGY"] = 123
+        ext.add_datasum()
+        ext.add_checksum()
+        hdul.append(ext)
+        hdul.writeto(testfile)
+
+        assert fitscheck.main([testfile]) == 1
+        expected = [
+            f"MISSING '{testfile}' .. Checksum not found in HDU #0",
+            f"MISSING '{testfile}' .. Datasum not found in HDU #0",
+            "2 errors",
+        ]
+        for exp, rec in zip(expected, caplog.records):
+            assert rec.message == exp
```

Comment:
looks like something is needed to make this more compatible with Windows?
https://github.com/astropy/astropy/actions/runs/17685106594/job/50268119896#step:10:1923